src/main/java/codeu/controller/RegisterServlet.java

```diff
@@ -1,20 +1,78 @@
 package codeu.controller;
+
 import java.io.IOException;
+
 import javax.servlet.ServletException;
+
 import javax.servlet.http.HttpServlet;
+
 import javax.servlet.http.HttpServletRequest;
+
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.mockito.Mockito;
+
+import codeu.model.data.User;
+import codeu.model.store.basic.UserStore;
+
+/** Servlet class responsible for user registration. */
+public class RegisterServlet extends HttpServlet {
 
-/**
-* Servlet class responsible for user registration.
-*/
-public class RegisterServlet extends HttpServlet
-{
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
-    throws IOException, ServletException
-  { 
-    // Forward request to register.jsp file
+      throws IOException, ServletException {
+
     request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
-  }//doGet
-}//RegisterServlet
+  } //doGet
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+
+    String username = request.getParameter("username");
+    String password = request.getParameter("password");
+
+    //Basic Error Handling
+
+    if (!username.matches("[\\w*\\s*]*")) {
+      request.setAttribute("error", "Please enter only letters, numbers, and spaces.");
+      request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
+      return;
+    } //if
+
+    //Check if username is already taken
+
```

Comment:
Same thing here, remove the extra line so readers can easily see what part of the code you're commenting on